vesin/src/vesin_cuda.cpp

```diff
@@ -80,47 +73,51 @@ vesin::cuda::get_cuda_extras(VesinNeighborList* neighbors) {
 }
 
 static void reset(VesinNeighborList& neighbors) {
+
     auto extras = vesin::cuda::get_cuda_extras(&neighbors);
 
-    if (neighbors.pairs && is_device_ptr(get_ptr_attributes(neighbors.pairs), "pairs")) {
-        CUDA_CHECK(cudaFree(neighbors.pairs));
+    if (neighbors.pairs && is_device_ptr(getPtrAttributes(neighbors.pairs), "pairs")) {
+        CUDART_SAFE_CALL(CUDART_INSTANCE.cudaFree(neighbors.pairs));
     }
-    if (neighbors.shifts && is_device_ptr(get_ptr_attributes(neighbors.shifts), "shifts")) {
-        CUDA_CHECK(cudaFree(neighbors.shifts));
+    if (neighbors.shifts && is_device_ptr(getPtrAttributes(neighbors.shifts), "shifts")) {
+        CUDART_SAFE_CALL(CUDART_INSTANCE.cudaFree(neighbors.shifts));
     }
-    if (neighbors.distances && is_device_ptr(get_ptr_attributes(neighbors.distances), "distances")) {
-        CUDA_CHECK(cudaFree(neighbors.distances));
+    if (neighbors.distances && is_device_ptr(getPtrAttributes(neighbors.distances), "distances")) {
+        CUDART_SAFE_CALL(CUDART_INSTANCE.cudaFree(neighbors.distances));
     }
-    if (neighbors.vectors && is_device_ptr(get_ptr_attributes(neighbors.vectors), "vectors")) {
-        CUDA_CHECK(cudaFree(neighbors.vectors));
+    if (neighbors.vectors && is_device_ptr(getPtrAttributes(neighbors.vectors), "vectors")) {
+        CUDART_SAFE_CALL(CUDART_INSTANCE.cudaFree(neighbors.vectors));
```

Comment:
random API suggestion: should this be 

```suggestion
        CUDART_CALL(cudaFree(neighbors.vectors));
```

(or some other name like `CUDART_DYNAMIC_CALL`, `GPULITE_CUDART_CALL`, …)? 

Basically hiding the global instance from users of gpu-lite